php-cs-fixer

```diff
@@ -83,7 +83,7 @@ if (class_exists('Phar')) {
 
 if ($require) {
     // OK, it's not, let give Composer autoloader a try!
-    $possibleFiles = [__DIR__.'/../../autoload.php', __DIR__.'/../autoload.php', __DIR__.'/vendor/autoload.php'];
+    $possibleFiles = [__DIR__.'/vendor/autoload.php', __DIR__.'/../../autoload.php'];
```

Comment:
`__DIR__.'/../autoload.php'` should never be needed

Reply:
> upper directories must be checked later

Why "must"? Previously first check was for most often use case and got hit, so others were not used, after your change first one is path used only in this repository, so everyone using PHP CS Fixer will attempt 1 `file_exists` which will always fail.

> `__DIR__.'/../autoload.php'` should never be needed

See the PR when it was added to see why it is needed: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4490

Reply:
> 
> 
> > upper directories must be checked later
> 
> Why "must"?

Easy, if `open_basedir` is set only to this repo, then trying to access any upper directory first will show php warning as the dir is outside.

> Previously first check was for most often use case and got hit, so others were not used, after your change first one is path used only in this repository, so everyone using PHP CS Fixer will attempt 1 `file_exists` which will always fail.

One or two extra file checks, nothing else negative, right?


> > `__DIR__.'/../autoload.php'` should never be needed
> 
> See the PR when it was added to see why it is needed: #4490

I will add this back.

Reply:
> if `open_basedir` is set only to this repo

When is that? Was this situation ever reported by anyone? Without any regression check this change makes no sense to me.

> One or two extra file checks, nothing else negative, right?

No one will die, but what is the point if we know that for anyone using PHP CS Fixer as library this check is useless?

Reply:
> When is that?

`php -d open_basedir="<current dir>" php-cs-fixer fix` (`open_basedir` is usually set with `php.ini` file, but the effect is the same)

Reply:
I know how to do it - I meant when it happens in real life? The example you gave means it's either for this repo or someone must have a `php-cs-fixer` file copied to his project's root.

For 1st case there is no `open_basedir` set.
For 2nd case I'd say it's incorrect usage (not sure that even work).

Reply:
1st case, maby 1st and halt ;-) because of:

> See the PR when it was added to see why it is needed: #4490